tests/countries/test_sint_maarten.py

```diff
@@ -0,0 +1,190 @@
+        for year in range(2015, 2077):
+            base_date = date(year, 10, 1)
+            second_monday = base_date + timedelta(days=(7 - base_date.weekday()) % 7 + 7)
+            self.assertHolidayName(name, second_monday)
+
+    def test_all_saints_day(self):
+        name = "All Saints' Day"
+        self.assertNoHolidayName(name, range(1900, 2011))
+        self.assertHolidayName(name, (f"{year}-11-01" for year in range(2011, 2077)))
+
+    def test_sint_maarten_day(self):
+        name = "Sint Maarten Day"
+        self.assertNoHolidayName(name, range(1900, 2011))
+        self.assertHolidayName(name, (f"{year}-11-11" for year in range(2011, 2077)))
+
+    def test_kings_day(self):
+        name = "King's Day"
+        self.assertNoHolidayName(name, range(1900, 2014))
+        self.assertHolidayName(
+            name,
+            "2014-04-26",  # Sunday, observed on Saturday
+            "2015-04-27",  # Monday
+            "2017-04-27",  # Thursday
+            "2019-04-27",  # Saturday
+            "2020-04-27",  # Monday
+            "2024-04-27",  # Saturday
+            "2025-04-26",  # Sunday, observed on Saturday
+        )
+        self.assertNoHoliday(
+            "2013-04-27",  # No King's Day
+            "2014-04-27",  # No Sunday observance
+            "2025-04-27",  # No Sunday observance
+        )
+
+    def test_kings_day_shifted_sunday(self):
+        name = "King's Day"
+        self.assertHolidayName(name, "2014-04-26")  # 2014: April 27 is Sunday
+        self.assertHolidayName(name, "2025-04-26")  # 2025: April 27 is Sunday
+        self.assertNoHoliday("2014-04-27")
+        self.assertNoHoliday("2025-04-27")
+
+    def test_kingdom_day(self):
+        name = "Kingdom Day"
+        # Check that Kingdom Day is not observed before 2011
+        self.assertNoHolidayName(name, range(1900, 2011))
+
+        # Check Kingdom Day from 2011 onward
+        for year in range(2011, 2077):
+            expected_date = get_expected_kingdom_day(year)
+            expected_date_str = expected_date.strftime("%Y-%m-%d")
+            self.assertHolidayName(name, expected_date_str)
+            # If the holiday is moved to Dec 16, ensure Dec 15 is not a holiday
+            if expected_date.day == 16:  # Moved to Monday
+                self.assertNoHolidayName(name, f"{year}-12-15")
+
+    def test_carnival_day(self):
+        name = "Carnival Day"
+        # Check that Carnival Day is not observed before 2011
+        self.assertNoHolidayName(name, range(1900, 2011))
+
+        # Check Carnival Day from 2011 onward
+        for year in range(2011, 2077):
+            expected_date = get_expected_carnival_day(year)
+            expected_date_str = expected_date.strftime("%Y-%m-%d")
+            self.assertHolidayName(name, expected_date_str)
+            # Optionally, ensure the holiday doesn't appear on other dates around April 30
+            for offset in [-2, -1, 1, 2]:  # Check nearby dates
```

Comment:
I'm not sure if this test is needed. If you are going to keep it -- use tuple.
```suggestion
            for offset in (-2, -1, 1, 2):  # Check nearby dates
```